File: discover_javascript/manipulate_the_data.js
var https = require('https');

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
        'User-Agent': 'Holberton_School',
        'Authorization': 'token ' + process.env.TOKEN
    }
};

var req = https.get(options, function (res) {
    const chunks = [];
    var jsonString = '';
    res.on('data', (chunk) => {
        chunks.push(chunk);
    });
    res.on('end', () => {
        jsonString = chunks.join('');
        var jsonobject = JSON.parse(jsonString);
        /*console.log(jsonobject.items);*/
        var full_names = jsonobject.items.map(function(obj){
            return obj.full_name;
        });

        console.log(full_names.join('\n'));
    });
});

req.on('error', function (e) {
    console.error(e);
});

File: discover_ruby/manipulate_the_data.rb
require 'httpclient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton School',
  'Authorization' => 'token f11ca8ea6bf7bf5456f862debf389d94961bdb15'
}

request = HTTPClient.new
results = request.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', query = nil, extheaders)
result_array = JSON.parse(results)
puts result_array['items'].map{|name| name['full_name']}.join("\n")
